fwutil/lib.py

```diff
@@ -687,8 +791,147 @@ def update_firmware(self, chassis_name, module_name, component_name):
             log_helper.log_fw_update_end(component_path, firmware_path, False, e)
             raise
 
+    def update_au_status_file(self, au_info_data, filename=FW_AU_STATUS_FILE_PATH):
+        with open(filename, 'w') as f:
+            json.dump(au_info_data, f)
+
+    def read_au_status_file_if_exists(self, filename=FW_AU_STATUS_FILE_PATH):
+        data = None
+        if os.path.exists(filename):
+            with open(filename) as au_status_file:
+                data = json.load(au_status_file)
+        return data
+
+    def set_firmware_auto_update_status(self, component_path, boot, rt_code):
+        data = self.read_au_status_file_if_exists(FW_AU_STATUS_FILE_PATH)
+        if data is None:
+            data = {}
+        if boot not in data:
+            data[boot] = []
+
+        au_status = data[boot]
+
+        comp_au_status = {}
+        if rt_code < -1:
+            status = False
+        else:
+            status = True
+
+        if rt_code == 0:
+            info = "reserved"
+        elif rt_code == 1:
+            info = "installed"
+        elif rt_code == 2:
+            info = "updated"
+        elif rt_code == 3:
+            info = "scheduled"
+        elif rt_code == -1:
+            info = "err_boot_type"
+        elif rt_code == -2:
+            info = "err_image"
+        elif rt_code == -3:
+            info = "err_others"
+        else:
+            info = "err_unknown"
+
+        comp_au_status['comp'] = component_path
+        comp_au_status['status'] = status
+        comp_au_status['info'] = info
+
+        au_status.append(comp_au_status)
+
+        self.update_au_status_file(data, FW_AU_STATUS_FILE_PATH)
+        return (status, info)
+
+    def get_au_status(self):
+        au_status = []
+        auto_updated_status_table = []
+        data = self.read_au_status_file_if_exists(FW_AU_STATUS_FILE_PATH)
+
+        if data is None:
+            return None
+
+        boot_type = list(data.keys())[0]
+        click.echo("Firmware auto-update performed for {} reboot".format(boot_type))
+
+        au_status = data[boot_type]
+        for comp_au_status in au_status:
+            r = []
+            r.append(comp_au_status['comp'] if 'comp' in comp_au_status else "")
+            r.append(comp_au_status['status'] if 'status' in comp_au_status else "")
+            r.append(comp_au_status['info'] if 'info' in comp_au_status else "")
+            auto_updated_status_table.append(r)
+
+        return tabulate(auto_updated_status_table, self.AU_STATUS_HEADER, tablefmt=self.FORMAT)
```

Comment:
@sujinmkang: This method is a duplicate of  'get_au_status' method in `ComponentStatusProvider` and is unused.